### PR TITLE
openstack-ardana: add update test automation playbooks

### PR DESCRIPTION
**Pull request by stefannica**
Adds playbooks automating the test procedure for
installing package updates on all the cloud nodes.

Also adds a playbook that can be used to reboot
all nodes in the correct sequence: deployer, then
controllers, then computes.

Depends on: 
 * #2503 
 * https://gerrit.suse.provo.cloud//#/c/3906/ (helper update playbooks)

### CODE REVIEW

**Comment by dirkmueller** [comment]
https://ci.suse.de/job/openstack-ardana/1255/console rebased and tested here- fails due to missing gerrit review

**Comment by stefannica** [comment]
> https://ci.suse.de/job/openstack-ardana/1255/console rebased and tested here- fails due to missing gerrit review

I installed the 3906 manually on the node and I kicked off a manual update run. Let's see if it passes.

**Comment by stefannica** [comment]
New job: https://ci.suse.de/job/openstack-ardana/1271/

**Comment by stefannica** [comment]
> New job: https://ci.suse.de/job/openstack-ardana/1271/

Turns out we need to start at least from devel cloudsource, otherwise we're missing the dac-3cp input model. To start from pool/GM cloudsource, we need an input model that has either 1 or 3 controller nodes, to keep mariadb happy.

Restarted the job with std-3cp instead of dac-3cp: https://ci.suse.de/job/openstack-ardana/1274/

**Comment by stefannica** [comment]
>> https://ci.suse.de/job/openstack-ardana/1255/console rebased and tested here- fails due to missing gerrit review
> I installed the 3906 manually on the node and I kicked off a manual update run. Let's see if it passes.

Running the update manually on the std-min setup failed in the spark pre-update status check while updating the second controller node (after successfully updating the deployer and updating/rebooting the first controller).  The spark-master service is down, probably because of https://issues.apache.org/jira/browse/SPARK-15544.

**Comment by stefannica** [comment]
Had to enable vendor changes to allow packages to be updated to D:C:8/:S. New job here: https://ci.suse.de/job/openstack-ardana/1289

File: scripts/jenkins/ardana/ansible/update-node.yml [hunk 1]
```diff
@@ -0,0 +1,141 @@
+---
+# Need to enable allowVendorChange to allow packages to be updated to D:C:8 and D:C:8:S
+- name: Enable allowVendorChange on the host
+  shell: |
```
Comment (dirkmueller) [inline scripts/jenkins/ardana/ansible/update-node.yml]: maybe use lineinfile: ansible module rather than shell'ing out here

**Pull request merged**